airflow/providers/amazon/aws/operators/s3_delete_objects.py

```diff
@@ -56,22 +59,34 @@ class S3DeleteObjectsOperator(BaseOperator):
     :type verify: bool or str
     """
 
-    template_fields = ('keys', 'bucket')
+    template_fields = ('keys', 'bucket', 'prefix')
 
     @apply_defaults
     def __init__(
             self,
             bucket,
-            keys,
+            keys=None,
+            prefix=None,
             aws_conn_id='aws_default',
             verify=None,
             *args, **kwargs):
+
+        if not keys and not prefix:
+            raise ValueError("Either keys or prefix should be set. Both are None")
+
         super().__init__(*args, **kwargs)
         self.bucket = bucket
         self.keys = keys
+        self.prefix = prefix
         self.aws_conn_id = aws_conn_id
         self.verify = verify
 
     def execute(self, context):
         s3_hook = S3Hook(aws_conn_id=self.aws_conn_id, verify=self.verify)
-        s3_hook.delete_objects(bucket=self.bucket, keys=self.keys)
+
+        if self.keys:
+            keys = self.keys
+        else:
+            keys = s3_hook.list_keys(bucket_name=self.bucket, prefix=self.prefix)
```

Comment:
```suggestion
        keys = self.keys or s3_hook.list_keys(bucket_name=self.bucket, prefix=self.prefix)
```

Reply:
I'm going to compare it to None as well as https://github.com/apache/airflow/pull/9350/files#r441908023.

```python
if self.keys is not None:
    keys = self.keys
else:
    ...
```

Reply:
The `or` operator does this. 👍 

See: https://realpython.com/python-or-operator/#using-or-with-common-objects

Reply:
It did not work well with the `or` operator when comparing with `is not None`.

```python
>>> a = None
>>> b = "test"
>>> a is not None or b
'test'
>>> a = "not None"
>>> a is not None or b
True
```

I fixed it as an if statement.

Reply:
```
>>> a = None
>>> b = "test"
>>> a or b
'test'
a = "not None"
a or b
'not None'
```
That's what you want, isn't it? You don't need to check for `is not None`.

Reply:
Passing an empty list to the S3DeleteObjectsOperator keys in the current implementation does not raise an exception. We had better not change this behavior. 

```python
keys = []
# Nothing is done. No exceptions will be made.
S3DeleteObjectsOperator(task_id=task_id, bucket=bucket, keys=keys)
```

If the argument is `keys=[]` and `prefix=None`, then the following code will execute `list_keys` with `prefix=None` and raise an exception.

```python
keys = self.keys or s3_hook.list_keys(bucket_name=self.bucket, prefix=self.prefix)
```

Reply:
> If the argument is keys=[] and prefix=None, then the following code will execute list_keys with prefix=None and raise an exception.

This case can't occur, because you added the validation in the `__init__` which makes sure that at least one of them is specified.

Reply:
I stopped considering an empty list and incorporated suggested changes. 👍  59f2e9c